bulldozer/merge.go

```diff
@@ -195,3 +197,30 @@ func allCommits(ctx context.Context, pullCtx pull.Context, client *github.Client
 
 	return repositoryCommits, nil
 }
+
+func postCommentIfUserActionable(ctx context.Context, client *github.Client, owner, repo string, prNum int, githubError github.ErrorResponse) {
+	commentBody := "Error occurred when attempting merge: " + githubError.Message
+
+	switch {
+	case strings.HasPrefix(githubError.Message, "You're not authorized to push to this branch."):
+		commentBody = commentBody + "\n\nThe bulldozer bot account may not be allowed to merge the branch."
+
+	case githubError.Message == "Merge commits are not allowed on this repository.":
+		commentBody = commentBody + "\n\nThe bulldozer configuration for merge method may not match the repository settings."
+
+	default:
+		// not user-actionable, don't post anything
+		return
+	}
+
+	logger := zerolog.Ctx(ctx)
+	logger.Debug().Msgf("Merge error may be user-fixable, commenting with: %q", commentBody)
+
+	if _, _, err := client.Issues.CreateComment(ctx, owner, repo, prNum, &github.IssueComment{Body: &commentBody}); err != nil {
```

Comment:
I think we need to check for any comments by our user or comments with same body to avoid duplicating the warning multiple times on a PR, if other events happen before the problem is resolved (e.g. branch updates, additional non-required status checks reporting, etc.)

Reply:
def need to check for previous comments or this can get spammy real quick